dipy/reconst/sfm.py

```diff
@@ -51,13 +49,21 @@
 # extensible, by inheriting from IsotropicModel/IsotropicFit below:
 
 # First, a helper function to derive the fit signal for these models:
-def _to_fit_iso(data, gtab):
-    data_no_b0 = data[..., ~gtab.b0s_mask]
+def _to_fit_iso(data, gtab, mask=None):
+    if mask is None:
+        mask = np.ones(data.shape[:-1], dtype=bool)
+    # Turn it into a 2D thing:
+    if len(mask.shape) > 0:
```

Comment:
I do not understand this part. Can you tell us more about this part?

Reply:
Yes: this is just to deal with the (rather unusual) case that your data is a one-dimensional vector representing one voxel's worth of data. In that case, `data.shape[:-1]` is  `()` and you might run into trouble here.

Reply:
ok, I see, thanks. I wonder if there is a better way to do that but It is fine like this too